Automattic/jetpack
# RNA: New layout structure for Container and Col

Closes #22466 

#### Changes proposed in this Pull Request:

* Refactor `Container` to include new props:
  *  `fluid` - It makes the container takes the entire available size, without adding vertical margins. Useful for compositions.
  * `horizontalSpacing` - The number of spacing (multiplied per 8px, ex: `2 spacing = 16px`) to be added in top and bottom of the container, default to 1.
  * `horizontalGutter` - The number of spacing (following same 8px rule) that will be added between the rows.
* Refactor `Col` to avoid adding more columns than expected and do not use `base-styles`.
* Refactor `AdminPage` to use the new approach of `Container` and `Col`.
* Refactor `AdminSection` and `AdminSectionHero` to not use `Container`.
* Refactor `Admin` of `Backup` to not use `Row`.
* Refactor `MyJetpack` components that use `Container` and `Col`:
  * `ProductCardsSection`
  * `MyJetpackScreen`

#### Jetpack product discussion



Described in #22466 

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






`For Container and Col`

* Goes to storybook and starts it.
* Navigate to `Playground / Layout`
* Interact with `controls`, it will allow you to test the current usages.

`For MyJetpack`

* Run the project and navigate to the `MyJetpack` page.
* You need to see all places using `grid layout` and matching correct spaces from `Figma`.

`For Backup`

* Build backup with `jetpack build plugins/backup`
* Navigate to the `Backup` menu item.
* You should see it normally.

#### Demo

_Container and Col_


https://user-images.githubusercontent.com/1663717/151620986-ad0b01f8-6800-4e17-ba9b-148430283a04.mp4

_Backup_


https://user-images.githubusercontent.com/1663717/151621014-c807fa77-1f0c-4e76-adf6-8e07fe48ba07.mp4

_My Jetpack_


https://user-images.githubusercontent.com/1663717/151621035-724968f3-88df-4713-85ab-a418b25135e7.mp4
